### ISSUE TITLE
Why does this compile!!??

### ISSUE THREAD

**Issue by blendsdk** [Microsoft/TypeScript#3489]
Hi,

I was wondering why the code below compiles fine!
The compiler should complain for the last two lines, not?

```typescript
interface ViewConfig {
	width?:number;
	height?:number;
}

interface ToolbarConfig {
	size?:number
}

class View implements ViewConfig {
	width:number;
	height:number;
	constructor(config:ViewConfig) {
		
	}
}

class Toolbar extends View implements ToolbarConfig {
	size:number;
	constructor(config:ToolbarConfig) {
		super(config);
	}
}


var myToolbar = new Toolbar('should not compile');
var myToolbar = new Toolbar(9999);
```

**Comment by carltonf** [comment]
Aren't all properties in the interface are optional? The it's ok that they are not there.
See http://www.typescriptlang.org/Handbook#interfaces-optional-properties

Quote: "The advantage of optional properties is that you can describe these possibly available properties while still also catching properties that you know are not expected to be available."

**Comment by RyanCavanaugh** [comment]
It's a structural type system, so an interface with all optional properties is assignable from anything.

This is a known pain point; see #391.
